docs/topics/gradle/gradle-compilation-and-caches.md

```diff
@@ -0,0 +1,340 @@
+[//]: # (title: Compilation and caches)
+
+On this page, you can learn about the following topics:
+* [Incremental compilation](#incremental-compilation)
+* [Gradle build cache support](#gradle-build-cache-support)
+* [Gradle configuration cache support](#gradle-configuration-cache-support)
+* [Build reports](#build-reports)
+* [Kotlin daemon and using it with Gradle](#kotlin-daemon-and-using-it-with-gradle)
+* [Defining Kotlin compiler execution strategy](#defining-kotlin-compiler-execution-strategy)
+
+## Incremental compilation
+
+The Kotlin Gradle plugin supports incremental compilation. Incremental compilation tracks changes to source files between
+builds so only files affected by these changes are compiled.
+
+Incremental compilation is supported for Kotlin/JVM and Kotlin/JS projects and is enabled by default.
+
+There are several ways to switch off incremental compilation:
+
+* `kotlin.incremental=false` for Kotlin/JVM.
+* `kotlin.incremental.js=false` for Kotlin/JS projects.
+* Use `-Pkotlin.incremental=false` or `-Pkotlin.incremental.js=false` as a command line parameter.
+
+  The parameter should be added to each subsequent build, and any build with incremental
+  compilation disabled invalidates incremental caches.
+
+The first build is never incremental.
+
+> Sometimes problems with incremental compilation become visible several rounds after the failure occurs. Use [build reports](#build-reports)
+> to track the history of changes and compilations. Doing so may also help you provide reproducible bug reports.
+>
+{type="tip"}
+
+### A new approach to incremental compilation
+
+> The new approach to incremental compilation is [Experimental](components-stability.md). It may be dropped or changed at any time.
```

Comment:
But we're not talking about incremental compilation in general, we're talking about the new approach.

Reply:
Good to know! It wasn't clear whether there was already incremental compilation being used before this "new approach". Incremental compilation itself could be considered a "new approach". Could you add a sentence to clarify that? Something like: In Kotlin we've been using incremental compilation since <version>. Since <newer version> we have a new experimental approach to incremental compilation.

Reply:
> Incremental compilation itself could be considered a "new approach". 

No:) There's a section about it above the section about the new approach.

Both of us haven't noticed the note at the bottom of this section:

> The new approach to incremental compilation is available since Kotlin 1.7.0 for the JVM backend in the Gradle
> build system only.
>
{type="note"}

Reply:
I suppose it wasn't said in the beginning because there would be two notes:

<img width="876" alt="Screenshot 2022-11-07 at 16 48 51" src="https://user-images.githubusercontent.com/78360457/200353827-c6c3522b-1965-4fff-a623-06f016c7bda8.png">

I moved this note to the first paragraph like an ordinary sentence.